packages/effect/src/internal/core-effect.ts

```diff
@@ -948,6 +948,30 @@ export const logAnnotations: Effect.Effect<HashMap.HashMap<string, unknown>> = c
     core.currentLogAnnotations
   )
 
+/** @internal */
+export const whenLogLevel = dual<
+  (
+    level: LogLevel.LogLevel | LogLevel.Literal
+  ) => <A, E, R>(effect: Effect.Effect<A, E, R>) => Effect.Effect<Option.Option<A>, E, R>,
+  <A, E, R>(
+    effect: Effect.Effect<A, E, R>,
+    level: LogLevel.LogLevel | LogLevel.Literal
+  ) => Effect.Effect<Option.Option<A>, E, R>
+>(2, (effect, level) => {
+  const requiredLogLevel = typeof level === "string" ? LogLevel.fromLiteral(level) : level
+
+  return core.withFiberRuntime((fiberState) => {
+    const minimumLogLevel = fiberState.getFiberRef(FiberRef.currentMinimumLogLevel)
```

Comment:
```suggestion
    const minimumLogLevel = fiberState.getFiberRef(fiberRuntime.currentMinimumLogLevel)
```

Will need to use the internal API here